packages/button/src/Button/components/GenericButton.tsx

```diff
@@ -169,6 +169,12 @@ const renderButton = (
     disableTabFocusAndIUnderstandTheAccessibilityImplications,
     onFocus,
     onBlur,
+    form,
+    formAction,
+    formMethod,
+    formEncType,
+    formTarget,
+    formNoValidate,
     ...rest
   } = props
   const customProps = getCustomProps(rest)
```

Comment:
In an ideal world, we'd probably get rid of this and allow the button to extend all native attributes but I think that's part of a bigger refactor instead of adding a filter for kebab case